app/src/main/java/com/mathroule/demoapp/MainActivity.java

```diff
@@ -28,6 +30,22 @@ public void onClick(View view) {
                         .setAction("Action", null).show();
             }
         });
+
+        try {
+            String text = Helper.getRandomButtonLabel();
+            ((Button) findViewById(R.id.buton)).setText(text);
+
+            if (text == "something") {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mathroule/demo-app/blob/ce05f89fe219cd29a1f984acb1a36cac9c595dea/app/src/main/java/com/mathroule/demoapp/MainActivity.java#L38"></span>Comparison of String objects using == or != in com.mathroule.demoapp.MainActivity.onCreate(Bundle)</td>
    </tr>
  </tbody>
</table>